datafusion/functions/src/regex/regexplike.rs

```diff
@@ -75,13 +75,8 @@ impl ScalarUDFImpl for RegexpLikeFunc {
         use DataType::*;
 
         Ok(match &arg_types[0] {
-            LargeUtf8 | Utf8 => Boolean,
             Null => Null,
-            other => {
-                return plan_err!(
-                    "The regexp_like function can only accept strings. Got {other}"
-                );
-            }
+            _ => Boolean,
```

Comment:
Without this change, regexp_like fails with an error about unsupported types if passed a StringView

With this change it succeeds (though the StringView is cast to a String)

@tshauck and I are going to organize a bunch of changes to get StringView working across functions as part of https://github.com/apache/datafusion/issues/11790

Reply:
Shouldn't we return error for non-string type?

`LargeUtf8 | Utf8 | Utf8View => Boolean`
`Null => Null`
`other => not supported`

Reply:
I was thinking the type coercion already handles, in theory, ensuring the correct types are passed in, so the extra check was not necessary. I can put back the explicit check if you prefer

Reply:
Seems #11618 is striking again. I think @alamb is right, but it is very counter intuitive to return a Boolean because that case shouldn't exists.

Should we use 'unreachable!' ?

Reply:
I think you are right, we don't need another type check in `return_type`.

> it is very counter intuitive to return a Boolean because that case shouldn't exists.

The type check should be handled in signature before `return_type`. We could improve docs about how the types and length check are handled in UDF/UDAF

Reply:
How about I'll at least add some comments to make it clear why this isn't doing the typechecks here

Reply:
> Should we use 'unreachable!' ?

I prefer using internal error when possible as `unreachable!` will panic. But now that you mention this, the function is going to panic at runtime anyways 🤔

Reply:
My bad I mixed up two things. In the thread about better error handling I think we agreed to return a planning error in this scenario, and some form of unreachable within the invoke function to signal that branch should never be executed

Reply:
I added comments in ead4a7c48c4789825f9d529e676e4fc169f3efa8